processor/dnslookupprocessor/README.md

```diff
@@ -0,0 +1,77 @@
+# DNSLookup Processor
+
+<!-- status autogenerated section -->
+| Status        |           |
+| ------------- |-----------|
+| Stability     | [development]: traces, metrics, logs   |
+| Distributions | [] |
+| Issues        | [![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3Aprocessor%2Fdnslookup%20&label=open&color=orange&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aopen+is%3Aissue+label%3Aprocessor%2Fdnslookup) [![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3Aprocessor%2Fdnslookup%20&label=closed&color=blue&logo=opentelemetry)](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aclosed+is%3Aissue+label%3Aprocessor%2Fdnslookup) |
+| Code coverage | [![codecov](https://codecov.io/github/open-telemetry/opentelemetry-collector-contrib/graph/main/badge.svg?component=processor_dnslookup)](https://app.codecov.io/gh/open-telemetry/opentelemetry-collector-contrib/tree/main/?components%5B0%5D=processor_dnslookup&displayType=list) |
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | [@andrzej-stencel](https://www.github.com/andrzej-stencel), [@kaisecheng](https://www.github.com/kaisecheng), [@edmocosta](https://www.github.com/edmocosta) |
+
+[development]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/component-stability.md#development
+<!-- end autogenerated section -->
+
+## Overview
+
+The `dnslookupprocessor` performs forward (hostname to IP) and/or reverse (IP to hostname) DNS lookups, enriching the target attribute with the resolved DNS data. It caches results to reduce expensive network lookups. The processor supports multiple host files and DNS servers for resolution.
+
+## Resolution Order and Retry Mechanism
+TODO
+
+## Caching
+TODO
+
+## Config
+
+Some notable settings include:
+
+- `context` (default: `resource`): Allows specifying the underlying telemetry context the processor will work with. Available values:
+  - `resource`: Resource attributes.
+  - `record`: Attributes within a data point, log record or a span.
+- `source_attributes`: An array of attribute names, which are used for lookup. The first valid hostname/IP is used.
```

Comment:
Can we add the configuration options `source_attributes` and `target_attribute` in a separate PR? For this PR, to simplify it, you could always use `source.address` for host name and `network.peer.address` for IP?